src/Workspaces/SharedUtilitiesAndExtensions/Workspace/CSharp/Utilities/SyntaxKindSet.cs

```diff
@@ -56,10 +56,12 @@ internal class SyntaxKindSet
                 SyntaxKind.VolatileKeyword,
             };
 
-        public static readonly ISet<SyntaxKind> LocalFunctionModifiers = new HashSet<SyntaxKind>(SyntaxFacts.EqualityComparer)
+        public static readonly ISet<SyntaxKind> AllLocalFunctionModifiers = new HashSet<SyntaxKind>(SyntaxFacts.EqualityComparer)
         {
             SyntaxKind.AsyncKeyword,
-            SyntaxKind.UnsafeKeyword
+            SyntaxKind.UnsafeKeyword,
+            SyntaxKind.ExternKeyword,
+            SyntaxKind.StaticKeyword
         };
```

Comment:
why do we need these two different sets?  i.e. we have AllLocalFunctionModifiers  here and s_validLocalFunctionModifiers  above.  Why can't they be the same...

Reply:
I thought the purpose of the individual sets was to indicate which keywords are valid in combination with a particular keyword. e.g. for ExternKeywordRecommender, we say that only 'static' and 'unsafe' are valid in combination with 'extern' on local functions.

Reply:
that's fair.  Though we often don't feel the need to be *that* picky.  i.e. it's not going to kill anyone to have `extern async`.  They'll simply get a message afterwards saying it's not legal.

What we *do* really want to avoid is *missing* keywords that are legal.  That's bad as that can interfere with typing.  Having some *extra* keywords in rare cases is just not that big a deal.